cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -190,20 +212,32 @@ func getStaticPodControlPlaneHashes(client clientset.Interface, nodeName string)
 
 	mirrorPodHashes := map[string]string{}
 	for _, component := range constants.MasterComponents {
-		staticPodName := fmt.Sprintf("%s-%s", component, nodeName)
-		staticPod, err := client.CoreV1().Pods(metav1.NamespaceSystem).Get(staticPodName, metav1.GetOptions{})
+		hash, err := getStaticPodSingleHash(client, nodeName, component)
 		if err != nil {
 			return nil, err
 		}
+		mirrorPodHashes[component] = hash
+	}
+	return mirrorPodHashes, nil
+}
 
-		podBytes, err := json.Marshal(staticPod)
-		if err != nil {
-			return nil, err
-		}
+// getStaticSinglePodHash computes hashes for a single Static Pod resource
+func getStaticPodSingleHash(client clientset.Interface, nodeName string, component string) (string, error) {
 
-		mirrorPodHashes[component] = fmt.Sprintf("%x", sha256.Sum256(podBytes))
+	mirrorPodHash := ""
+	staticPodName := fmt.Sprintf("%s-%s", component, nodeName)
+	staticPod, err := client.CoreV1().Pods(metav1.NamespaceSystem).Get(staticPodName, metav1.GetOptions{})
+	if err != nil {
+		return mirrorPodHash, err
 	}
-	return mirrorPodHashes, nil
+
+	podBytes, err := json.Marshal(staticPod)
+	if err != nil {
+		return mirrorPodHash, err
+	}
+
+	mirrorPodHash = fmt.Sprintf("%x", sha256.Sum256(podBytes))
```

Comment:
maybe just return the `fmt.Sprintf` directly?

Reply:
done